src/ucp/wireup/wireup.c

```diff
@@ -738,7 +736,19 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
     status = ucp_wireup_select_lanes(ep, params, ep_init_flags, address_count,
                                      address_list, addr_indices, &key);
     if (status != UCS_OK) {
-        goto err;
+        if (ep->flags & UCP_EP_FLAG_USED) {
+            failed_lane = 0; /* lanes aren't selected but need to invoke err
+                                handler, lane number doesn't matter here */
+            goto err;
+        }
+
+        for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
```

Comment:
why don't we call ucp_worker_set_ep_failed() if ep is not used?
maybe ucp_worker_set_ep_failed() could handle this case as well?